Sources/UtilityBeltNetworking/HTTPClient.swift

```diff
@@ -19,15 +19,21 @@ public class HTTPClient {
 
     /// The URLSession that is used for all requests.
     private let session: URLSession
+    
+    /// The protocol to handle session delegate responses, if needed.
+    private weak var sessionDelegate: HTTPSessionDelegate?
 
     // MARK: - Methods
 
     // MARK: Initializers
 
     /// Initializes a new HTTPClient with a given URLSession.
-    /// - Parameter session: The URLSession to use for all requests.
-    public init(session: URLSession = .shared) {
+    /// - Parameters:
+    ///   - session: The URLSession to use for all requests.
+    ///   - delegate: [optional] The session to handle the responses, if needed.
+    public init(session: URLSession = .shared, delegate: HTTPSessionDelegate? = nil) {
```

Comment:
Can we make this an additional `internal` `init` constructor so that way we can make `HTTPSessionDelegate` and `HTTPBackgroundSessionDelegate` internal as well, leaving the original init exposed to `public` for consumers?